plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1529,6 +1529,13 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
     {
         IcebergColumnHandle handle = (IcebergColumnHandle) column;
         Table icebergTable = catalog.loadTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName());
+        Schema schema = icebergTable.schema();
+        icebergTable.spec().fields().stream()
+                .filter(field -> schema.findField(field.sourceId()).name().equals(handle.getName()))
```

Comment:
You should do an `id` based comparison here, rather than name. `field.sourceId() == handle.getId()`

Reply:
Thank @alexjo2144  for the review. Addressed comments!